spec/models/spree/order/state_machine_spec.rb

```diff
@@ -162,5 +153,12 @@
       allow(order).to receive_messages state: "canceled"
       allow(order).to receive_messages allow_resume?: true
     end
+
+    it "should set payment state to 'balance due'" do
+      expect {
+        order.resume!
```

Comment:
@mkllnk been fighting with this spec tonight and I am not sure what I am doing wrong. I expect when I `order.resume!`, the checkout state should run the after_transition block [here](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/models/spree/order/checkout.rb#L45) and change the state to `resumed` and then calling order.after_resume. For some reason on the test, it only sets the state to `canceled` which causes the test to fail. Any ideas that I may have overlooked?

Reply:
You can go in with a debug statement like `binding.pry` to find out if the `save` is success in that method. And if not, you can call `errors` to find out why not. The test data may be incomplete.

Reply:
After some digging, it seems like when you set the state in the `before do` column via
`allow(order).to receive_messages state: "canceled"`, this unfortunately is being set again in the after_transaction block. So the transition to "resume" is successful and in the after_transaction block, order.state is referenced as "canceled" and setting order.state = order.state, sets it back to canceled. 

Not sure why we introduced the after_transaction block here tho. The state is already set but looks like we are re-setting it again. Loved this bug hunt, got a good understanding about rails state machines 😄